### PR TITLE
[NRLF-505] Reporting RDS

### PR DESCRIPTION
**Pull request by brianbayes**

### CODE REVIEW

File: terraform/infrastructure/modules/rds/vpc.tf [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+}
+
+#------------------------------------------------------------------------------
+# Subnets
+#------------------------------------------------------------------------------
+
+resource "aws_subnet" "public_subnet" {
+  count                   = var.subnet_count.public
+  vpc_id                  = aws_vpc.vpc.id
+  cidr_block              = element(var.public_subnet_cidr_blocks, count.index)
```
Comment (Rohoolio) [inline terraform/infrastructure/modules/rds/vpc.tf:39]: Ive not seen the use of element before, thats quite snazzy

File: README.md [hunk 2]
```diff
@@ -569,3 +571,61 @@ When you create a release branch in the form of `release/yyyy-mm-dd` or `hotfix/
...
 
 This is because it will use that value to tag the commit once its been merged into main as a reference point, and this is how it tracks which release it is as github actions struggles with post merge branch identification
+
+## Bastion
+
+For Dev and Test there is a bastion in place that can be used to connect to the RDS instance either through the CLI or from your chosen database client for Postgres.
+
+There are a few steps to get this sorted:
+
+1. Create a .pem file and adding it to your ssh agent:
```
Comment (Rohoolio) [inline README.md:581]: Only comment is if you make the key pair in AWS that it auto downloads for you which my smooth brain didnt realise haha - Also can you download a keypair once its been created? if not how would you get the persistent dev one if its already been made?

Comment (brianbayes) [reply README.md:581]: I have created [NRLF-558](https://nhsd-jira.digital.nhs.uk/browse/NRLF-558) which will improve the process of storing a pem file for non-prod environments.

File: terraform/infrastructure/modules/reporting/secrets.tf [hunk 3]
```diff
@@ -0,0 +1,18 @@
+resource "random_password" "rds_password" {
+  length  = 17
+  special = false
+}
+
+resource "aws_secretsmanager_secret" "rds_password" {
+  name                    = "${replace(var.prefix, "--", "-")}-${var.name}-rds"
+  description             = "Password for ${replace(var.prefix, "--", "-")}-${var.name}-rds"
+  recovery_window_in_days = var.prefix == "prod" ? 30 : 0
```
Comment (nomad3k) [inline terraform/infrastructure/modules/reporting/secrets.tf:9]: This value must be between 7 and 30, so 0 is not valid.

File: terraform/infrastructure/modules/reporting/ec2.tf [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+      echo "Finished installing psql"
+    EOF
+
+  tags = {
+    "Name" = "bastion-${var.environment}"
+  }
+}
+
+
+data "aws_key_pair" "bastion_key" {
```
Comment (nomad3k) [inline terraform/infrastructure/modules/reporting/ec2.tf:22]: This line is throwing this error when I do a plan. 
image

**Review by nomad3k — CHANGES_REQUESTED** [review]

File: terraform/infrastructure/modules/reporting/secrets.tf (hunk 3 above)
Comment (brianbayes) [reply terraform/infrastructure/modules/reporting/secrets.tf:9]: Not quite, you can set this to 0 in terraform as per this: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/secretsmanager_secret#recovery_window_in_days

File: README.md (hunk 2 above)
Comment (brianbayes) [reply README.md:583]: I've also left instructions in the README that you can just set the `bastion_enabled` to `false` which will void the creation of the bastion for local dev.

File: terraform/infrastructure/modules/reporting/ec2.tf (hunk 4 above)
Comment (brianbayes) [reply terraform/infrastructure/modules/reporting/ec2.tf:22]: I've added some instructions to the README for this, you can just set the `bastion_enabled` variable to false which will void the need to create a key pair in EC2. I've also created [NRLF-588](https://nhsd-jira.digital.nhs.uk/browse/NRLF-558) to visit the possibility of using S3 to store the `.pem` file in the future.

**Pull request closed without merging**